src/Fantomas.Core.Tests/RecordTests.fs

```diff
@@ -2076,3 +2076,48 @@ and seekReadParamExtras (retRes, paramsRes) (idx: int) =
                 Attributes = enum<ParameterAttributes> flags
                 CustomAttrs = cas }
 """
+
+[<Test>]
+let ``equality comparison with a `with` expression should format correctly with default alignment, 2507`` () =
+    formatSourceString
+        false
+        """
+let compareThings (first: Thing) (second: Thing) =
+    first = { second with
+                Foo = first.Foo
+                Bar = first.Bar
+            }
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+let compareThings (first: Thing) (second: Thing) =
+    first = { second with
+                Foo = first.Foo
+                Bar = first.Bar }
+"""
+
+[<Test>]
```

Comment:
Not the end of the world, but I would typically not add an extra test for `{ config with MultilineBlockBracketsOnSameColumn = true }`. 
The problem was present with the default configuration and should have been reported as such.
There is no more code coverage by this additional test.

Just a heads up for the future.
What would be a more interesting test case would be switching the order of the arguments in the infix application.